# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/PServiceWorkerRegistration.ipdl
# Commit: 185488c90cb8
# Full Hash: 185488c90cb8d3cd7f7041bc457f5e5cbb1617b2
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2021-06-10 15:47:45
# Regressor Bug: 1564235
# File Overlap Count: 2
# Description:
#   Bug 1564235 - P3 Create IPC for NavigationPreload in PServiceWorkerRegistration.ipdl. r=dom-worker-reviewers,asuth
#   
#   This patch implements the IPC for NavigationPreload APIs.
#   The IPC methods are created on PServiceWorkerRegistration.
#   
# ==============================================================================

diff -r afae3c50f7ac -r 185488c90cb8 dom/serviceworkers/PServiceWorkerRegistration.ipdl
--- a/dom/serviceworkers/PServiceWorkerRegistration.ipdl	Thu Jun 10 06:01:26 2021 +0000
+++ b/dom/serviceworkers/PServiceWorkerRegistration.ipdl	Thu Jun 10 06:01:27 2021 +0000
@@ -4,6 +4,7 @@
 
 include protocol PBackground;
 
+include IPCNavigationPreloadState;
 include IPCServiceWorkerRegistrationDescriptor;
 
 include "ipc/ErrorIPCUtils.h";
@@ -22,6 +23,11 @@
   async Update(nsCString aNewestWorkerScriptUrl) returns (
     IPCServiceWorkerRegistrationDescriptorOrCopyableErrorResult aResult);
 
+  // For NavigationPreload interface
+  async SetNavigationPreloadEnabled(bool aEnabled) returns (bool aSuccess);
+  async SetNavigationPreloadHeader(nsCString aHeader) returns (bool aSuccess);
+  async GetNavigationPreloadState() returns (IPCNavigationPreloadState? aState);
+
 child:
   async __delete__();
 